### DISCUSSION TITLE
GetItemById -- can't seem to get the field values.

### DISCUSSION THREAD

**Discussion by juliemturner** [pnp/pnpcore#533] [Q&A] [answered]
So, sadly... I'm stuck again.  I just want to get an item by id.  I can get the item, but none of the field values so I can't do anything with it.  

var changeList = pnpContext.Web.Lists.GetById(new Guid(notification.Resource), p => p.Title, p => p.Items, p => p.Fields.QueryProperties(p => p.InternalName,
                                                                                    p => p.FieldTypeKind,
                                                                                    p => p.TypeAsString,
                                                                                    p => p.Title));

var item = changeList.Items.GetById((change as IChangeItem).ItemId);

But item["Version"] and item["My_Field_Name"] etc throw an exception because they're not populated.  I can't seem to find any better documentation on what piece I'm missing.

**Comment by jansenbe** [comment]
https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html#getting-list-item-versions will help with list item versions and also shows how to request all list item fields (results in $select=*). This should get you all the needed data, if you don't see the fields you need in the Values collection you can also cross check the `FieldValuesAsText` collection.

So your code would something like this:

```csharp
var item = changeList.Items.GetById((change as IChangeItem).ItemId, p => p.All, p => p.Versions);
```

**Reply by juliemturner** [reply]
I had tried that first, but probably not in combination with the extra request on the list part... However, I've now tried it again with p => p.All 
`var item = changeList.Items.GetById((change as IChangeItem).ItemId, p => p.All);`
and I'm still getting:

```
'item["Current_x0020_Version"]' threw an exception of type 'PnP.Core.ClientException'
    Data: {System.Collections.ListDictionaryInternal}
    Error: {PnP.Core.ClientError}
    HResult: -2146233088
    HelpLink: null
    InnerException: null
    Message: "Property Current_x0020_Version does not exist in this entity"
    Source: "PnP.Core"
    StackTrace: "   at PnP.Core.Model.ExpandoBaseDataModel`1.get_Item(String key)"
    TargetSite: {System.Object get_Item(System.String)}
```

Current_x0020_Version is just a single line of text field... so it's nothing special.  I dont' really want versions I just want the Version property field... to give me the current version number. Regardless that would be step 2, at this point I can't get the simple thing to work. I grant you I'm probably doing it wrong, but I can't see what the issue is.

Also, FieldValuesAsText has a count of 0.  

In case it's useful this is what item looks like:

```
{PnP.Core.Model.SharePoint.ListItem}
    All: null
    AttachmentFiles: {}
    Comments: {}
    CommentsDisabled: '((PnP.Core.Model.SharePoint.ListItem)item).CommentsDisabled' threw an exception of type 'PnP.Core.ClientException'
    CommentsDisabledScope: '((PnP.Core.Model.SharePoint.ListItem)item).CommentsDisabledScope' threw an exception of type 'PnP.Core.ClientException'
    ContentType: {PnP.Core.Model.SharePoint.ContentType}
    Count: 17
    FieldValuesAsHtml: {PnP.Core.Model.SharePoint.FieldStringValues}
    FieldValuesAsText: {PnP.Core.Model.SharePoint.FieldStringValues}
    FieldValuesForEdit: {PnP.Core.Model.SharePoint.FieldStringValues}
    File: {PnP.Core.Model.SharePoint.File}
    FileSystemObjectType: File
    Folder: {PnP.Core.Model.SharePoint.Folder}
    HasChanges: true
    Id: 2
    Key: 2
    LikedByInformation: {PnP.Core.Model.SharePoint.LikedByInformation}
    MappingHandler: {Method = {System.Object <.ctor>b__2_0(PnP.Core.Services.FromJson)}}
    Metadata: Count = 7
    Parent: {}
    ParentList: {PnP.Core.Model.SharePoint.List}
    PnPContext: {PnP.Core.Services.PnPContext}
    PostMappingHandler: {Method = {Void <.ctor>b__2_1(System.String)}}
    Properties: {PnP.Core.Model.SharePoint.PropertyValues}
    Requested: true
    RoleAssignments: {}
    ServerRedirectedEmbedUri: "https://redacted"
    ServerRedirectedEmbedUrl: "https://redacted"
    Title: null
    UniqueId: '((PnP.Core.Model.SharePoint.ListItem)item).UniqueId' threw an exception of type 'PnP.Core.ClientException'
    Values: Count = 17
    Versions: {}
    Dynamic View: Expanding the Dynamic View will get the dynamic members for the object
```

**Reply by jansenbe** [reply]
Hey @juliemturner , 

Can you cross check what's inside the values collection if you just load the item by id as shown in below code. I do see all my custom fields popping up here. Maybe you get this error because of the way how you write `Current_x0020_Version`... when I test this I need to use `Current Version` as name to get the property.

![image](https://user-images.githubusercontent.com/7451219/131506632-c4266e9b-ab00-4e83-979e-44966857a1e3.png)

**Reply by jansenbe** [reply]
Here's a screenshot showing fields with underscores and spaces in their names:
![image](https://user-images.githubusercontent.com/7451219/131508494-3eb961c7-e462-4ac2-8eb6-02096dae6e7d.png)

**Comment by juliemturner** [comment, marked as answer]
Ok, that solved it.  It was that there is an actual space, so it's the display name not the internal name...

**Reply by jansenbe** [reply]
I'll update the docs to clarify this
